File: codeforces/1800~1899/1898/d.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  int n = 0;
  cin >> n;
  vector<int> a(n), b(n);
  for (int i = 0; i < n; ++i) cin >> a[i];
  for (int i = 0; i < n; ++i) cin >> b[i];

  i64 base = 0;
  for (int i = 0; i < n; ++i) {
    base += abs(a[i] - b[i]);
  }

  vector<array<int,3>> p;
  for (int i = 0; i  < n; ++i) {
    int x1 = min(a[i], b[i]);
    int x2 = max(a[i], b[i]);
    p.push_back({x1, x2, i});
  }
  sort(p.begin(), p.end());

  i64 ans = base;
  priority_queue<array<int,3>, vector<array<int,3>>, greater<>> pq;
  for (int k = 0; k < n; ++k) {
    int i = p[k][2];
    if (!pq.empty()) {
      auto [x2, x1, j] = pq.top();
      if (x2 < p[k][0]) {
        i64 ans2 = base - (abs(a[j] - b[j]) + abs(a[i] - b[i])) +
          (abs(a[j] - b[i]) + abs(b[j] - a[i]));
        ans = max(ans, ans2);
      }
    }
    pq.push({p[k][1], p[k][0], i});
  }

  cout << ans << endl;
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1800~1899/1898/d.rs
use std::io::{stdin, stdout, BufWriter, Write};
use std::str::FromStr;
use std::fmt::Debug;
use std::cmp::{min, max};

fn read_int<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.trim().parse::<T>().expect("invalid input")
}

fn read_vec<T: FromStr>() -> Vec<T> where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.split(" ").map(|x| x.trim().parse().expect("invalid input")).collect()
}

fn solve() {
    let stdout = stdout();
    let mut writer = BufWriter::new(stdout.lock());

    let n = read_int::<usize>();
    let a = read_vec::<i64>();
    let b = read_vec::<i64>();
    let mut base: i64 = 0;
    for i in 0..n {
        base += (a[i] - b[i]).abs();
    }

    let mut p = Vec::new();
    for i in 0..n {
        let l = min(a[i], b[i]);
        let r = max(a[i], b[i]);
        p.push((l, r, i));
    }
    p.sort();

    let mut ans = base;
    let mut left: (i64, i64, usize) = p[0].clone();
    for i in 1..n {
        if left.1 < p[i].0 {
            let i1 = p[i].2;
            let i2 = left.2;
            ans = max(ans, base - (a[i1] - b[i1]).abs() - (a[i2] - b[i2]).abs() +
                (a[i1] - b[i2]).abs() + (a[i2] - b[i1]).abs());
        }
        if p[i].1 < left.1 {
            left = p[i].clone();
        }
    }

    writeln!(writer, "{}", ans).ok();
}

fn main() {
    let n = read_int::<i32>();
    for _t in 0..n {
        solve();
    }
}
